pkgs/development/python-modules/graphite-web/default.nix

```diff
@@ -61,49 +53,67 @@ buildPythonPackage rec {
     whisper
   ];
 
-  postPatch = ''
-    substituteInPlace setup.py \
-      --replace "Django>=1.8,<3.1" "Django" \
-      --replace "django-tagging==0.4.3" "django-tagging"
-  '';
+  pythonRelaxDeps = [
+    "django"
+    "django-tagging"
+  ];
 
   # Carbon-s default installation is /opt/graphite. This env variable ensures
   # carbon is installed as a regular Python module.
   GRAPHITE_NO_PREFIX = "True";
 
   preConfigure = ''
     substituteInPlace webapp/graphite/settings.py \
-      --replace "join(WEBAPP_DIR, 'content')" "join('$out', 'webapp', 'content')"
+      --replace-fail \
+        "join(WEBAPP_DIR, 'content')" \
+        "join('$out/webapp', 'content')"
   '';
 
-  checkInputs = [ mock ];
+  nativeCheckInputs = [
+    mock
+    rrdtool
+    writableTmpDirAsHomeHook
+  ];
+  preCheck =
+    # redis not practical in test environment
```

Comment:
It is! e.g. `pkgs/development/python-modules/django-q2/default.nix`

Reply:
And redisTestHook is on staging